### PR TITLE
Convert `VariancePopulation` to UDAF

### LINKED ISSUE #10668: Convert Variance Population to UDAF
### Is your feature request related to a problem or challenge?

Similar to #10667 but for Variance Population function

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mknaw**
## Which issue does this PR close?

Closes #10668

## Are these changes tested?

yes, via sqllogictest, I believe...

## Are there any user-facing changes?

sure hope not

### CODE REVIEW

File: datafusion/physical-expr/src/aggregate/build_in.rs [hunk 1]
```diff
@@ -709,44 +703,6 @@ mod tests {
...
-                    assert_eq!("c1", result_agg_phy_exprs.name());
-                    assert_eq!(
-                        Field::new("c1", DataType::Float64, true),
-                        result_agg_phy_exprs.field().unwrap()
-                    )
-                }
-            }
-        }
-        Ok(())
-    }
```
Comment (mknaw) [inline datafusion/physical-expr/src/aggregate/build_in.rs:748]: I _believe_ this is covered by the sqllogictests, but this belief may be ignorant

**Comment by jayzhan211** [comment]
I think you need to comment out `variance pop` in proto

https://github.com/apache/datafusion/blob/6b7021479c956ba3ca2a04fff81487cd57f80624/datafusion/proto/proto/datafusion.proto#L483

And, run `./regen.sh` under `datafusion/proto`

**Review by goldmedal — APPROVED** [review]
Thanks @mknaw. Overall LGTM.

**Review by jayzhan211 — APPROVED** [review]
👍

**Comment by jayzhan211** [comment]
Thanks @mknaw and @goldmedal

**Pull request merged**